### PR TITLE
Remove wireserver fallback for imds calls

### PR DESCRIPTION
**Pull request by narrieta**
The fallback to wireserver is no longer supported

### CODE REVIEW

File: tests/common/protocol/test_imds.py [hunk 1]
```diff
@@ -386,99 +386,86 @@ def test_endpoint_fallback(self):
         # http GET calls and enforces a single GET call (fallback would cause 2) and
         # checks the url called.
 
-        test_subject = imds.ImdsClient("foo.bar")
+        test_subject = imds.ImdsClient()
```
Comment (maddieford) [inline tests/common/protocol/test_imds.py:389]: We should probably change this test name + comment since we're no longer using a fallback method.

**Review by maddieford — APPROVED** [review]

File: azurelinuxagent/ga/update.py [hunk 2]
```diff
@@ -482,7 +482,7 @@ def _get_vm_size(self, protocol):
         """
         if self._vm_size is None:
 
-            imds_client = get_imds_client(protocol.get_endpoint())
+            imds_client = get_imds_client()
```
[the remark this answers is not in the document]
Comment (narrieta) [reply azurelinuxagent/ga/update.py]: thanks; i removed the entire method

[the remark this answers is not in the document]
Comment (maddieford) [reply azurelinuxagent/ga/update.py]: I didn't catch that it was being mocked in tests still:
tests.lib.tools.AgentTestCaseWithGetVmSizeMock.setUp

[the remark this answers is not in the document]
Comment (narrieta) [reply azurelinuxagent/ga/update.py]: Thanks, updated

**Review by maddieford — APPROVED** [review]

**Pull request merged**